### PR TITLE
Update GeoStylerContext.unsupportedProperties

### LINKED ISSUE #2172: Replace UnsupportedPropertiesContext with useGeoStylerUnsupportedProperties

### PR DESCRIPTION
**Pull request by KaiVolland**
This replaces the legacy `UnsupportedPropertiesContext` with the `GeoStylerContext`.

It also fixes the RuleTable example and adds an `unsupportedProperties` prop to the `GeoStylerContext` example:

localhost_6060_ (18)

closes #2172 

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

### CODE REVIEW

File: src/context/GeoStylerContext/GeoStylerContext.example.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ The `GeoStylerContext` lets you define translations, static component props, and
 #### Example
 
 Provide static component props. Here, we globally define the rendererType to use OpenLayers for rendering.
+And the unsupportedProperties. Usually you would use `unsupportedProperties` from a style parser like.
```
Comment (jansule) [inline src/context/GeoStylerContext/GeoStylerContext.example.md:36]: Can we write proper sentences here?

File: src/context/GeoStylerContext/GeoStylerContext.tsx [hunk 2]
```diff
@@ -56,10 +47,17 @@ export type CompositionContext = {
   Style?: StyleComposableProps;
 };
 
+export interface UnsupportedPropertiesContext extends UnsupportedProperties {
+  options?: {
+    hideUnsupported?: boolean;
+    locale?: GeoStylerLocale['UnsupportedPropertiesUtil'];
```
Comment (jansule) [inline src/context/GeoStylerContext/GeoStylerContext.tsx:53]: Why is this not part of the other locale?

File: src/context/GeoStylerContext/GeoStylerContext.tsx [hunk 3]
```diff
@@ -84,7 +82,26 @@ export const useGeoStylerLocale = <T extends keyof GeoStylerLocale>(key: T): Geo
   return structuredClone(ctx.locale[key]);
 };
 
-export const useGeoStylerUnsupportedProperties = (): any => {
+export const useGeoStylerUnsupportedProperties = <T extends Symbolizer>(symbolizer: T) => {
```
Comment (jansule) [inline src/context/GeoStylerContext/GeoStylerContext.tsx:85]: Do we have to extent this to `Style`? E.g. the props of a Rule might also differ between two parsers

File: src/context/GeoStylerContext/GeoStylerContext.tsx [hunk 4]
```diff
@@ -84,7 +82,26 @@ export const useGeoStylerLocale = <T extends keyof GeoStylerLocale>(key: T): Geo
...
   const ctx = useContext(GeoStylerContext);
-  return structuredClone(ctx.unsupportedProperties);
+
+  if (!ctx.unsupportedProperties) {
+    return {
+      getSupportProps: (propName: keyof T) => {}
```
Comment (jansule) [inline src/context/GeoStylerContext/GeoStylerContext.tsx:90]: Should this have the same structure as the return value at the end of the function? I.e.

```js
return {
    {},
    getSupportProps: (propName: keyof T) => {}
  };
```

**Review by jansule — CHANGES_REQUESTED** [review]

File: src/context/GeoStylerContext/GeoStylerContext.tsx (hunk 3 above)
Comment (KaiVolland) [reply src/context/GeoStylerContext/GeoStylerContext.tsx:85]: Correct. Currently this works for the Symbolizer Editors.

File: src/context/GeoStylerContext/GeoStylerContext.tsx (hunk 2 above)
Comment (KaiVolland) [reply src/context/GeoStylerContext/GeoStylerContext.tsx:53]: The "other" locale is not yet supported.

File: src/Util/UnsupportedPropertiesUtil.ts [hunk 5]
```diff
@@ -46,36 +45,39 @@ class UnsupportedPropertiesUtil {
...
   }): Partial<FormItemProps> => {
 
-    if (!unsupportedProperties || Object.keys(unsupportedProperties).length < 1) {
+    if (!context || Object.keys(context).length < 1) {
       return {};
     }
 
+    const {
+      hideUnsupported = false,
+      locale = en_US.UnsupportedPropertiesUtil
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/Util/UnsupportedPropertiesUtil.ts:61]: You can not use React stuff directly in this util.

File: src/context/GeoStylerContext/GeoStylerContext.example.md [hunk 6]
```diff
@@ -45,6 +47,27 @@ const GeoStylerContextExample = () => {
       Renderer: {
         rendererType: 'OpenLayers'
       }
+    },
+    unsupportedProperties: {
+      Symbolizer: {
+        'LineSymbolizer': {
```
Comment (jansule) [inline src/context/GeoStylerContext/GeoStylerContext.example.md:53]:
```suggestion
        LineSymbolizer: {
```

**Review by jansule — APPROVED** [review]
Very nice! Just one very minor remark. Feel free to merge afterwards.

**Pull request merged**